aviary/interface/methods_for_level2.py

```diff
@@ -1436,20 +1455,82 @@ def add_linkage_constraint(self, phase_a, phase_b, var_a, var_b, connected,
+
+                    for state in states_to_link:
+                        # in initial guesses, all of the states, other than time use the same name
+                        temp_state = 'times' if state == 'time' else state
+                        initial_guesses1 = self.phase_info[phase1]['initial_guesses']
+                        initial_guesses2 = self.phase_info[phase2]['initial_guesses']
+
+                        # if a state is in the initial guesses, get the units of the initial guess
+                        units1, units2 = True, True
+                        if temp_state in initial_guesses1:
+                            units1 = initial_guesses1[temp_state][-1]
+                        if temp_state in initial_guesses2:
+                            units2 = initial_guesses2[temp_state][-1]
+                        else:
+                            units1 = units2 = True
+
+                        # if neither phase is analytic and the units of the initial guesses match we can use link_phases
+                        if not (analytic1 or analytic2) and (units1 == units2):
+                            if state == 'alpha':
+                                # alpha is always connected with a constraint
+                                connected = False
+                            else:
+                                # we can't use connected=True if we are running in parallel, but we usually want it
+                                connected = true_unless_mpi
+                            self.traj.link_phases(
+                                [phase1, phase2], [state], connected=connected)
+
+                        # if either phase is analytic or the units of the intial guesses don't match, we have to use a linkage_constraint
+                        else:
+                            # analytic phases use the prefix "initial" for time and distance, but not mass
+                            if analytic2:
+                                prefix = 'initial_'
+                            else:
+                                prefix = ''
+                            add_linkage_constraint(self, phase1, phase2,
+                                                   'time', prefix+'time', True)
+                            add_linkage_constraint(self, phase1, phase2,
+                                                   'distance', prefix+'distance', True)
+                            add_linkage_constraint(self, phase1, phase2, 'mass',
+                                                   'mass', False, ref=1.0e5)
+
+                self.traj.link_phases(
+                    phases=[
+                        "ascent", "accel"], vars=[
+                        "time", "mass", Dynamic.Mission.VELOCITY], connected=True)
+
+                # add all params and promote them to self.model level
+                ParamPort.promote_params(
+                    self.model,
+                    trajs=["traj"],
+                    phases=[
```

Comment:
I think we might also need to add the reserve phases to this phase list in promote_params.